components/map/index.jsx

```diff
@@ -0,0 +1,162 @@
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+
+import Icon from '../icon';
+import Modal from '../modal';
+import { MAP } from '../../utilities/constants';
+
+const displayName = MAP;
+
+const propTypes = {
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Whether the modal containing the map is open. Only for `modal` variant
+	 */
+	isModalOpen: PropTypes.bool,
+	/**
+	 * Title for the Map component.
+	 */
+	title: PropTypes.string,
+	/**
+	 * Accepts a single Map location
+	 */
+	location: PropTypes.string,
+	/**
+	 * Accepts Map locations as an array of Objects
+	 */
+	locations: PropTypes.arrayOf(PropTypes.object),
+	/**
+	 * Accepts a Google Map API Key that will be used for showing the map
+	 */
+	googleAPIKey: PropTypes.string.isRequired,
+	/**
+	 * Accepts a node that will be shown at the footer of the modal containing the map. Only for `modal` variant
+	 */
+	modalFooter: PropTypes.node,
+	/**
+	 * Selects the variant of Map component
+	 */
+	variant: PropTypes.oneOf(['standalone', 'modal']),
+};
+
+const defaultProps = {
+	isModalOpen: true,
+	variant: 'standalone',
+};
+
+/**
+ * A map component is used to find a location
+ */
+class Map extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			selected: 0,
+		};
+		this.generatedId = shortid.generate();
+	}
+
+	/**
+	 * Get the Map's HTML id. Generate a new one if no ID present.
+	 */
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	render() {
+		const mapContainer = (
+			<div
+				id={this.getId()}
+				className={classNames(
+					`slds-grid`,
+					{'slds-has-coordinates' : this.props.locations},
+					this.props.className
+				)}
+			>
+				<div className="slds-map_container">
+					<div className="slds-map">
+						<iframe
+							id={`${this.getId()}-google-map`}
+							title="Google Maps iframe"
+							src={`https://www.google.com/maps/embed/v1/place?key=${
+								this.props.googleAPIKey
+							}&q=${ this.props.locations ?
+								encodeURIComponent(this.props.locations[this.state.selected].address)
+								: encodeURIComponent(this.props.location)
+							}`}
+						/>
+					</div>
+				</div>
+				{ this.props.locations ? (
+				<div className="slds-coordinates">
+					<div className="slds-coordinates__header">
+						<h2 className="slds-coordinates__title">{`${this.props.title} (${
+							this.props.locations.length
+						})`}</h2>
+					</div>
+					<ul className="slds-coordinates__list">
+						{this.props.locations.map((location, i) => (
+							<li key={location.name} className="slds-coordinates__item">
+								<span className="slds-assistive-text" aria-live="polite" />
```

Comment:
Please take a look at https://www.lightningdesignsystem.com/components/map/#First-Coordinate-Selected to understand why this is here. See:

`<span class="slds-assistive-text" aria-live="polite">Worldwide Corporate Headquarters is currently selected</span>`

Since it's a list of buttons and aria-pressed isn't handled well by all screen readers, assistive text is here.